lib/classes/PluginManager.js

```diff
@@ -0,0 +1,120 @@
+'use strict';
+
+const Utils = require('./Utils');
+const path = require('path');
+const has = require('lodash').has;
+const forEach = require('lodash').forEach;
+
+class PluginManager {
+  constructor(serverless) {
+    this._serverless = serverless;
+    this._pluginInstances = [];
+    this._commandsList = [];
+    this._commands = {};
+  }
```

Comment:
Personally I would switch to serverless instead of S. I don't really like having short names for variables as its always hard to understand. It took me a while to understand what S meant for example